### PR TITLE
Merge cffi branch and update

### PR DESCRIPTION
**Pull request by flacjacket**
Playing with this, I saw you had your cffi branch, I am definitely a big fan of that. This is a merge of that branch, in addition to a couple updates, in particular to use cffi's out-of-line compilation API from 1.0, using your socket context manager with iwlist, and not explicitly zeroing memory, since cffi does that by default.

One thing that will be new with the out-of-line cffi module is to run this from source, you need to run `python iwlib/_iwlib_build.py`, but setuptools has great hooks for doing all the cffi compilation during install.

### CODE REVIEW

**Comment by nathan-hoad** [comment]
Arg I've completely missed this, I'm so sorry! I'll take a look over the weekend and try to get it in and released, but this looks fantastic! Thank you.

File: README.rst [hunk 1]
```diff
@@ -27,6 +27,12 @@ good to go.
 Change History
 --------------
 
+1.6
+    - Added support for Python 3 (via migration to CFFI)
+    - **API INCOMPATABILITY**: Dictionaries in 1.5 for Python 3 returned keys
+      of type `str`, but values of types `bytes`. They now always return type
+      `bytes`.
```
Comment (nathan-hoad) [inline README.rst]: What would you think about me undoing this? It doesn't feel nice to reach into a dictionary with a byte string, it'd be nice to just use keys of type `str` everywhere, in all python versions.

Comment (flacjacket) [inline README.rst]: Yeah, that sounds good to me! I just pushed a change to do that, let me know if that looks good.

**Comment by nathan-hoad** [comment]
Awesome, thanks so much for this! I'm working on some reproducible testing for the library, I'm hoping to have something working soon.

**Pull request merged**